### PR TITLE
[forge] Add performance test and LBT suite

### PR DESCRIPTION
**Pull request by zihaoccc**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update or suggest changes to the docs at https://developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: testsuite/testcases/src/performance_test.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    fn run<'t>(&self, ctx: &mut NetworkContext<'t>) -> Result<()> {
+        let duration = Duration::from_secs(240);
+        let all_validators = ctx
+            .swarm()
+            .validators()
+            .map(|v| v.peer_id())
+            .collect::<Vec<_>>();
+
+        // Generate some traffic
+        let _txn_stat = generate_traffic(ctx, &all_validators, duration)?;
```
Comment (bmwill) [inline testsuite/testcases/src/performance_test.rs:26]: I would expect that we'd output the stats.

Although I don't particularly like the fact that dips in perf can fail the existing cluster test, in order to have feature parity we should probably support that. Perhaps by:

```rust
pub struct PerformanceBenchmark {
    tps_threshold: Option<u64>,
}
```

Where if tps_threshold is Some(x) that we fail if tps is lower that x

Comment (zihaoccc) [reply testsuite/testcases/src/performance_test.rs:26]: I will add threshold to indicate performance regression

Comment (zihaoccc) [reply testsuite/testcases/src/performance_test.rs:26]: As we discussed offline, perf fail detection will be added in next PR for ci-test change

Comment (bmwill) [reply testsuite/testcases/src/performance_test.rs:26]: Do we want to output the stats?

**Review by bmwill — APPROVED** [review]

**Review by bmwill — DISMISSED** [review]

**Pull request closed without merging**